### PR TITLE
Add a default_value_source property in HelpParameter

### PR DESCRIPTION
**Pull request by arrownj**
Previously, both default value from function default and `az configure` will all display as `Default:`, we want to divide these different source. Add this property to save the default value source. It will be displayed in the help message.

### CODE REVIEW

File: knack/commands.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ def _resolve_default_value_from_config_file(self, arg, overrides):
             logger.info("Configured default '%s' for arg %s", config_value, arg.name)
             overrides.settings['default'] = DefaultStr(config_value)
             overrides.settings['required'] = False
+            overrides.settings['default_from'] = 'Configure'
```
Comment (haroldrandom) [inline knack/commands.py:102]: What other else sources could be ?

Comment (arrownj) [reply knack/commands.py:102]: For local context, it's `Local Context`. If no value for this property, it will be treated as `Default`.

Comment (haroldrandom) [reply knack/commands.py:102]: What other values could be?

File: knack/parser.py [hunk 2]
```diff
@@ -174,6 +175,7 @@ def load_command_table(self, command_loader):
                 param.deprecate_info = arg.deprecate_info
                 param.preview_info = arg.preview_info
                 param.experimental_info = arg.experimental_info
+                param.default_from = arg.default_from
```
Comment (jiasli) [inline knack/parser.py:178]: I prefer calling it `default_value_source` for better understanding.

File: knack/commands.py (hunk 1 above)
Comment (arrownj) [reply knack/commands.py]: Currently, there are only 3 sources: Default, Configure, Local Context.

File: knack/parser.py (hunk 2 above)
Comment (arrownj) [reply knack/parser.py]: thanks, updated.

File: knack/commands.py [hunk 3]
```diff
@@ -99,6 +99,7 @@ def _resolve_default_value_from_config_file(self, arg, overrides):
             logger.info("Configured default '%s' for arg %s", config_value, arg.name)
             overrides.settings['default'] = DefaultStr(config_value)
             overrides.settings['required'] = False
+            overrides.settings['default_value_source'] = 'Configure'
```
Comment (jiasli) [inline knack/commands.py:102]: `Configure` is a verb. Maybe `config`?

```suggestion
            overrides.settings['default_value_source'] = 'config'
```

Comment (arrownj) [reply knack/commands.py:102]: This value will be shown in help message directly. The other two sources are `Default` and `Local Context`, so maybe `Config` is more reasonable ?

Comment (arrownj) [reply knack/commands.py]: Update to `Config`.

**Pull request merged**